src/app/system/system.service.ts

```diff
@@ -224,4 +224,49 @@ export class SystemService {
     return this.http.put(`/accountnumberformats/${accountNumberPreferenceId}`, accountNumberPreferenceChanges);
   }
 
+  /**
+   * @param {any} filterBy Properties by which entries should be filtered.
+   * @param {string} orderBy Property by which entries should be sorted.
+   * @param {string} sortOrder Sort order: ascending or descending.
+   * @param {number} offset Page offset.
+   * @param {number} limit Number of entries within the page.
+   * @returns {Observable<any>} Audit Trails.
+   */
+  getAuditTrailsWithOptions(filterBy: any, orderBy: string, sortOrder: string, offset: number, limit: number): Observable<any> {
+    let httpParams = new HttpParams()
```

Comment:
Ensure server side pagination and filtering works by making the required change as I specified above.